tests/ui/test_ui_tools.py

```diff
@@ -1235,6 +1235,44 @@ def test_main_view_generates_EDITED_label(self, mocker,
             assert label[0].text == '(EDITED)'
             assert label[1][1] == 11
 
+    @pytest.mark.parametrize('to_vary_in_each_message, editing_allowed,\
+                             pass_test', [
+        ({'sender_id': 2, 'timestamp': 45}, True, False),
+        ({'sender_id': 1, 'timestamp': 1}, True, False),
+        ({'sender_id': 1, 'timestamp': 45}, False, False),
+        ({'sender_id': 1, 'timestamp': 45}, True, True),
+    ], ids=['msg_sent_by_other_user',
+            'time_limit_esceeded',
+            'editing_not_allowed',
+            'all_conditions_met'])
+    def test_keypress_edit_message(self, mocker, message_fixture, pass_test,
+                                   to_vary_in_each_message, editing_allowed):
+        varied_message = dict(message_fixture, **to_vary_in_each_message)
+        key = 'e'
+        size = (20,)
+        msg_box = MessageBox(varied_message, self.model, message_fixture)
+        msg_box.model.user_id = 1
+        msg_box.model.initial_data = dict()
+        msg_box.model.initial_data[
+            'realm_allow_message_editing'] = editing_allowed
+        msg_box.model.initial_data[
+            'realm_message_content_edit_limit_seconds'] = 60
```

Comment:
Why not combine the previous lines to initialize `initial-data` in one statement?